packages/anu-vue/src/composables/useConfig.ts

```diff
@@ -0,0 +1,54 @@
+import type { MaybeRef } from '@vueuse/core'
+import type { App, InjectionKey, Ref } from 'vue'
+import { getCurrentInstance, provide } from 'vue'
+import { zIndexContextKey } from '@/composables/useZIndex'
+import type { ConfigProviderProps } from '@/components/config-provider/config-provider-props'
+
+export type ConfigProviderContext = Partial<ConfigProviderProps>
+export const configProviderContextKey: InjectionKey<
+  Ref<ConfigProviderContext>
+> = Symbol('AnuConfigProviderKey')
+
+// Storing this value is to be able to use useConfig outside of the component.
+const globalConfig = ref<ConfigProviderContext>()
+
+export function useConfig() {
+  return getCurrentInstance()
+    ? inject(configProviderContextKey, globalConfig)
+    : globalConfig
+}
+
+export function provideConfig(
+  config: MaybeRef<ConfigProviderProps>,
+  app?: App,
+) {
+  const provideFn = getProvideFunction(app)
+  if (!provideFn)
+    return
+
+  const context = computed(() => unref(config))
+
+  provideFn(configProviderContextKey, context)
+
+  provideFn(zIndexContextKey, computed(() => context.value.zIndex))
```

Comment:
Hey 👋🏻 

While I was working on `AConfig` (same as `AConfigProvider` but also for component props) I had issue where if I update the config it also get updated upwards in DOM tree due to reactivity. Then I asked antfu & he suggested me directly passing new ref in `provide` instead of updating the value. While I was reviewing this PR for new release, I notices you are doing the same thing to avoid updating config in DOM upwards of `AConfigProvider`, Am I right?

excalidraw graph: [props-defaults-explanation.excalidraw.zip](https://github.com/jd-solanki/anu/files/11723225/props-defaults-explanation.excalidraw.zip)

Reply:
I tried your tricked and it seems its working 🤯 

I'll merge AConfigProvider & AConfig in this PR if possible!